src/YTable.cc

```diff
@@ -146,6 +146,33 @@ YTable::hasMultiSelection() const
     return ! YSelectionWidget::enforceSingleSelection();
 }
 
+YItem *
+YTable::findItem( const std::string & wantedItemLabel, int column ) const
+{
+    return YTable::findItem( wantedItemLabel, column, itemsBegin(), itemsEnd() );
+}
+
+
+YItem *
+YTable::findItem( const std::string & wantedItemLabel,
+		  int                 column,        
+                  YItemConstIterator  begin,
+                  YItemConstIterator  end ) const
+{
+    if ( ! hasColumn( column ) )
+  	return 0;
+
+    for ( YItemConstIterator it = begin; it != end; ++it )
```

Comment:
Hm, I wanted to avoid iterators and ugly `for` loop with the ++ operator. But with `std::find_if` it does not look much better:

```c++
   auto iter = std::find_if(begin, end, [&](YItem *ti){
       auto item = dynamic_cast<YTableItem *>(ti);
       return item && item->label( column ) == wantedItemLabel;
    });

   return (iter == end) ? nullptr : *iter;
```

So I'm fine with the current solution if you find it better. Or maybe somebody else can suggest a better solution...

Reply:
well, I am not sure which C++ standard we use, but at least you can write using C++17 ( so quite new standard )
```
for( auto&& [begin, end]: it )
```

Reply:
We use C++11. I already tried to upgrade that few months ago to something newer. But then the libyui-*-pkg packages failed to compile (because of some compile issues with the libzypp headers).

We would need to compile only some libyui parts with the new version. And I'm not sure if that would actually work...

Reply:
BTW the C++ version is set [here](https://github.com/libyui/libyui/blob/9bd5aad6682f85870713190cb9ab8fa22e7953e6/buildtools/LibyuiCommon.cmake#L76).

Reply:
I think I'll just leave this as it is.